utils/common/src/main/java/brooklyn/util/time/Time.java

```diff
@@ -443,36 +500,466 @@ else if (s.equalsIgnoreCase("d") || s.equalsIgnoreCase("day") || s.equalsIgnoreC
+
+        return returnResult;
+    }
+
+    @SuppressWarnings("deprecation")
+    private static Maybe<Calendar> parseCalendarDefaultParse(String input) {
+        try {
+            long ms = Date.parse(input);
+            if (ms>=new Date(1999, 12, 25).getTime() && ms <= new Date(2200, 1, 2).getTime()) {
+                // accept default date parse for this century and next
+                GregorianCalendar c = new GregorianCalendar();
+                c.setTimeInMillis(ms);
+                return Maybe.of((Calendar)c);
+            }
+        } catch (Exception e) {
+            Exceptions.propagateIfFatal(e);
+        }
+        return Maybe.absent();
+    }
+
+    private static Maybe<Calendar> parseCalendarUtc(String input) {
+        input = input.trim();
+        if (input.matches("\\d+")) {
+            if ("0".equals(input)) {
+                // accept 0 as epoch UTC
+                return Maybe.of(newCalendarFromMillisSinceEpochUtc(0));
+            }
+            Maybe<Calendar> result = Maybe.of(newCalendarFromMillisSinceEpochUtc(Long.parseLong(input)));
+            if (result.isPresent()) {
+                int year = result.get().get(Calendar.YEAR);
+                if (year >= 2000 && year < 2200) {
+                    // only applicable for dates in this century
+                    return result;
+                } else {
+                    return Maybe.absent("long is probably not millis since epoch UTC; millis as string is not in acceptable range");
+                }
+            }
+        }
+        return Maybe.absent("not long millis since epoch UTC");
+    }
+
+    private final static String DIGIT = "\\d";
+    private final static String LETTER = "\\p{L}";
+    private final static String COMMON_SEPARATORS = "-\\.";
+    private final static String TIME_SEPARATOR = COMMON_SEPARATORS+":";
+    private final static String DATE_SEPARATOR = COMMON_SEPARATORS+"/ ";
+    private final static String DATE_TIME_ANY_ORDER_GROUP_SEPARATOR = COMMON_SEPARATORS+":/ ";
+
+    private final static String DATE_ONLY_WITH_INNER_SEPARATORS = 
+            namedGroup("year", DIGIT+DIGIT+DIGIT+DIGIT)
+            + anyChar(DATE_SEPARATOR)
+            + namedGroup("month", options(optionally(DIGIT)+DIGIT, anyChar(LETTER)+"+"))
+            + anyChar(DATE_SEPARATOR)
+            + namedGroup("day", optionally(DIGIT)+DIGIT);
+    private final static String DATE_WORDS_2 = 
+        namedGroup("month", anyChar(LETTER)+"+")
+        + anyChar(DATE_SEPARATOR)
+        + namedGroup("day", optionally(DIGIT)+DIGIT)
+        + ",?"+anyChar(DATE_SEPARATOR)+"+"
+        + namedGroup("year", DIGIT+DIGIT+DIGIT+DIGIT);
+    // we could parse NN-NN-NNNN as DD-MM-YYYY always, but could be confusing for MM-DD-YYYY oriented people, so require month named
+    private final static String DATE_WORDS_3 = 
+        namedGroup("day", optionally(DIGIT)+DIGIT)
+        + anyChar(DATE_SEPARATOR)
+        + namedGroup("month", anyChar(LETTER)+"+")
```

Comment:
This formatting (starting the line with a +) is a bit weird to me. It's a bit inconsistent between patterns at the moment, too (e.g. `TIME_ONLY_WITH_INNER_SEPARATORS` is at the end of the line).